src/main.rs

```diff
@@ -93,46 +102,299 @@ macro_rules! create_socket {
+        eth_phy_nrst.set_high().unwrap();
+        eth_phy_nrst.set_low().unwrap();
+        eth_phy_nrst.set_high().unwrap();
+        let _rmii_ref_clk = gpioa.pa1.into_alternate_af11().set_speed(hal::gpio::Speed::VeryHigh);
+        let _rmii_mdio = gpioa.pa2.into_alternate_af11().set_speed(hal::gpio::Speed::VeryHigh);
+        let _rmii_mdc = gpioc.pc1.into_alternate_af11().set_speed(hal::gpio::Speed::VeryHigh);
+        let _rmii_crs_dv = gpioa.pa7.into_alternate_af11().set_speed(hal::gpio::Speed::VeryHigh);
+        let _rmii_rxd0 = gpioc.pc4.into_alternate_af11().set_speed(hal::gpio::Speed::VeryHigh);
+        let _rmii_rxd1 = gpioc.pc5.into_alternate_af11().set_speed(hal::gpio::Speed::VeryHigh);
+        let _rmii_tx_en = gpiob.pb11.into_alternate_af11().set_speed(hal::gpio::Speed::VeryHigh);
+        let _rmii_txd0 = gpiob.pb12.into_alternate_af11().set_speed(hal::gpio::Speed::VeryHigh);
+        let _rmii_txd1 = gpiog.pg14.into_alternate_af11().set_speed(hal::gpio::Speed::VeryHigh);
+
+        // TODO: Configure the ethernet controller
+        // Enable the ethernet peripheral.
+        //clocks.apb4.enr().modify(|_, w| w.syscfgen().set_bit());
+        //clocks.ahb1.enr().modify(|_, w| {
+        //    w.eth1macen().set_bit()
+        //        .eth1txen().set_bit()
+        //        .eth1rxen().set_bit()
+        //});
+
+        //dp.SYSCFG.pmcr.modify(|_, w| unsafe { w.epis().bits(0b100) }); // RMII
+
+        cp.SCB.enable_icache();
 
         init_log();
         // info!("Version {} {}", build_info::PKG_VERSION, build_info::GIT_VERSION.unwrap());
         // info!("Built on {}", build_info::BUILT_TIME_UTC);
         // info!("{} {}", build_info::RUSTC_VERSION, build_info::TARGET);
 
-        // c.schedule.tick(Instant::now()).unwrap();
+        let mut debug_pin = gpioc.pc6.into_push_pull_output();
+        debug_pin.set_low().unwrap();
+
+        let mut dac_pin = gpiob.pb15.into_push_pull_output();
+        dac_pin.set_low().unwrap();
+
+        // Configure timer 2 to trigger conversions for the ADC
+        let mut timer2 = dp.TIM2.timer(500.khz(), &mut clocks);
+        timer2.listen(hal::timer::Event::TimeOut);
 
-        let dp = c.device;
         init::LateResources {
-            spi: (dp.SPI1, dp.SPI2, dp.SPI4, dp.SPI5),
-            i2c: dp.I2C2,
-            ethernet_periph: (
-                dp.ETHERNET_MAC,
-                dp.ETHERNET_DMA,
-                dp.ETHERNET_MTL,
-            ),
+            adc1: adc1_spi,
+            dac1: dac1_spi,
+            adc2: adc2_spi,
+            dac2: dac2_spi,
+
+            dbg_pin: debug_pin,
+            dac_pin: dac_pin,
+            timer: timer2,
+
+            _eeprom_i2c: i2c2,
+//            ethernet_periph: (
+//                dp.ETHERNET_MAC,
+//                dp.ETHERNET_DMA,
+//                dp.ETHERNET_MTL,
+//            ),
+        }
+    }
+
+    #[task(binds = TIM2, resources = [dbg_pin, timer, adc1, adc2])]
+    fn tim2(mut c: tim2::Context) {
```

Comment:
Note that this costs about 30-40 cycles (100 ns per servo cycle) compared to DMA. I.e. 5-10% more load. That's significant. OK for the transition phase but should be changed later.

Reply:
Yeah, there's a TODO in the original PR indicating conversion to utilizing DMA again. I was seeing some bizarre behavior at first, so I went for the simpler solution for the time being.

Reply:
Yes. Good approach.

Reply:
DMA has now been implemented for triggering channel updates.